backend/routes/fellows.js

```diff
@@ -60,93 +60,87 @@ const readFellowByEmail = async (req, res, next) => {
-
-//     const response = await queries.editFellow({
-//         fellowId,
-//         firstname,
-//         lastname,
-//         picture,
-//         bio,
-//         linkedIn,
-//         github,
-//         cohortId,
-//         wantMentor
-//     });
-//     res.status(201);
-//     res.json({
-//         status: "success",
-//         message: `fellow.${fellowId} data updated`,
-//         payload: response
-//     });
-//   } catch (err) {
-//     handleError(err, req, res, next);
-//   }
-// };
+const createFellow = async (req, res, next) => {
+  try {
+    const fFirstName = processInput(req.body.fFirstName, "hardVC", "first name", 30);
+    const fLastName = processInput(req.body.fLastName, "hardVC", "last name", 30);
+    const fEmail = processInput(req.body.fEmail, "hardVC", "fellow email", 50);
+    const cohortId = processInput(req.body.cohort_id, "idNum", "cohort id");
+
+    const response = await queries.addFellow({
+        fFirstName,
+        fLastName,
+        fEmail,
+        cohortId
+    });
+    res.status(201);
+    res.json({
+        status: "success",
+        message: `new fellow '${fFirstName} ${fLastName}' added`,
+        payload: response
+    });
+  } catch (err) {
+    handleError(err, req, res, next);
+  }
+};
 
+const updateFellow = async (req, res, next) => {
+  try {
+    const fId = processInput(req.params.id, "idNum", "fellow id");
+    const fFirstName = processInput(req.body.fFirstName, "hardVC", "first name", 30);
+    const fLastName = processInput(req.body.fLastName, "hardVC", "last name", 30);
+    // const fEmail = processInput(req.body.fEmail, "hardVC", "fellow email", 50);
+    const fPicture = processInput(req.body.fPicture, "softVC", "picture url", undefined);
+    const fBio = processInput(req.body.fBio, "softVC", "bio", undefined);
+    const fLinkedIn = processInput(req.body.fLinkedIn, "softVC", "linkedin url", 150);
+    const fGithub = processInput(req.body.fGithub, "softVC", "github url", 150);
+    const wantMentor = processInput(req.body.wantMentor, "bool", "want mentor bool");
+
+    const response = await queries.editFellow({
+        fId,
+        fFirstName,
+        fLastName,
+        // fEmail,
+        fPicture,
+        fBio,
+        fLinkedIn,
+        fGithub,
+        wantMentor
+    });
+    res.status(200);
+    res.json({
+        status: "success",
+        message: `fellow.${fId} data has been updated`,
+        payload: response
+    });
+  } catch (err) {
+    handleError(err, req, res, next);
+  }
+};
 
-/*
-// f_id SERIAL PRIMARY KEY,
-f_first_name VARCHAR (30) NOT NULL,
-f_last_name VARCHAR (30) NOT NULL,
-// f_email VARCHAR (50) REFERENCES users_data(user_email),
-f_picture VARCHAR,
-f_bio VARCHAR,
-f_linkedin VARCHAR (150),
-f_github VARCHAR (150),
-cohort INT REFERENCES classes(class_id),
-want_mentor BOOLEAN NOT NULL DEFAULT FALSE
-*/
+const deleteFellow = async (req, res, next) => {
```

Comment:
Has been implemented in routes/auth.js

Reply:
deleteFellow block in routes deactivated